packages/block-editor/src/components/block-list/use-in-between-inserter.js

```diff
@@ -135,9 +138,18 @@ export function useInBetweenInserter() {
 					return;
 				}
 
-				// Don't show the inserter when hovering above (conflicts with
-				// block toolbar) or inside selected block(s).
-				if ( getSelectedBlockClientIds().includes( clientId ) ) {
+				// Don't show the inserter if the following conditions are met,
+				// as it conflicts with the block toolbar:
+				// 1. when hovering above or inside selected block(s)
+				// 2. when the orientation is vertical
+				// 3. when the __experimentalCaptureToolbars is not enabled
+				// 4. when the Top Toolbar is not disabled
+				if (
+					getSelectedBlockClientIds().includes( clientId ) &&
```

Comment:
I wonder if we should still show the inserter and maybe hide the block toolbar if it conflicts. It's quite annoying that you can use the in-between inserter above the selected block.

Reply:
Right, is there no selector for the currently shown indicator that we can add there? Not saying it should be done in this PR, could be a follow-up

Reply:
It is possible to simply not show the block toolbar when the inserter is displayed via the `isBlockInsertionPointVisible` selector, but this might be a little confusing as the block toolbar will flash frequently:

https://github.com/user-attachments/assets/3c5653bb-07f7-4d74-8fd2-440303a8eb03

Ideally, the block toolbar should only not appear when there is a visual "conflict", but I don't know how to do that.

Reply:
Right, should only be hidden when the inserter is above the selected blocks. I believe there's an insertion index and root client ID in the store that we can check?

Reply:
Should we merge this and follow up with different solutions for "vertical" layouts?

Reply:
Sure:
> Not saying it should be done in this PR, could be a follow-up

Reply:
Thank you, I will merge this PR 👍